File: internal/query/cost.go
package query

type ModelCallCost struct {
	CallID         string `json:"callId"`
	IdentityBasis  string `json:"identityBasis"`
	Basis          string `json:"basis"`
	AmountMicroUSD *int64 `json:"amountMicroUsd,omitempty"`
	PrimaryReason  string `json:"primaryReason,omitempty"`
	RateEntryID    string `json:"rateEntryId,omitempty"`
}

type ModelCallRef struct {
	CallID        string `json:"callId"`
	IdentityBasis string `json:"identityBasis"`
	EvidenceRole  string `json:"evidenceRole"`
}

type CostReasonCount struct {
	Reason string `json:"reason"`
	Count  int64  `json:"count"`
}

type CostSummary struct {
	AmountMicroUSD  *int64            `json:"amountMicroUsd,omitempty"`
	Basis           string            `json:"basis"`
	Coverage        string            `json:"coverage"`
	EligibleCalls   int64             `json:"eligibleCalls"`
	PricedCalls     int64             `json:"pricedCalls"`
	UnpricedReasons []CostReasonCount `json:"unpricedReasons"`
	AggregateError  string            `json:"aggregateError,omitempty"`
}

File: web/src/model/cost.ts
export type CostBasis = "provider_reported" | "rate_card_estimate" | "mixed" | "unavailable" | "unknown";
export type CostCoverage = "complete" | "partial" | "unavailable" | "unknown";
export type CostReasonCount = Readonly<{ reason: string; count: bigint }>;
export type CostSummary = Readonly<{
  amountMicroUsd: bigint | null;
  basis: CostBasis;
  coverage: CostCoverage;
  eligibleCalls: bigint;
  pricedCalls: bigint;
  unpricedReasons: readonly CostReasonCount[];
  aggregateError?: string;
}>;

export type ModelCallCost = Readonly<{
  callId: string; identityBasis: string; basis: CostBasis; amountMicroUsd: bigint | null;
  primaryReason?: string; rateEntryId?: string;
}>;

export type ModelCallRef = Readonly<{ callId: string; identityBasis: string; evidenceRole: string }>;
